clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackProviderFactory.groovy

```diff
@@ -16,42 +16,20 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.client
 
-import com.netflix.spinnaker.clouddriver.openstack.security.AccountType
 import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackNamedAccountCredentials
-import org.openstack4j.api.OSClient
-import org.openstack4j.core.transport.Config
-import org.openstack4j.model.common.Identifier
-import org.openstack4j.openstack.OSFactory
 
 /**
  * Builds the appropriate {@link OpenstackClientProvider} based on the configuration.
  */
 class OpenstackProviderFactory {
 
   static OpenstackClientProvider createProvider(OpenstackNamedAccountCredentials credentials) {
-    OSClient osClient
-    OpenstackClientProvider provider
-    Config config = credentials.insecure ? Config.newConfig().withSSLVerificationDisabled() : Config.newConfig()
-    if (AccountType.V2.value() == credentials.accountType) {
-      osClient = OSFactory.builderV2()
-        .withConfig(config)
-        .endpoint(credentials.endpoint)
-        .credentials(credentials.username, credentials.password)
-        .tenantName(credentials.tenantName)
-        .authenticate()
-      provider = new OpenstackClientV2Provider(osClient, credentials.regions)
-    } else if (AccountType.V3.value() == credentials.accountType) {
-      osClient = OSFactory.builderV3()
-        .withConfig(config)
-        .endpoint(credentials.endpoint)
-        .credentials(credentials.username, credentials.password, Identifier.byName(credentials.domainName))
-        .scopeToProject(Identifier.byName(credentials.tenantName), Identifier.byName(credentials.domainName))
-        .authenticate()
-      provider = new OpenstackClientV3Provider(osClient)
-    } else {
-      throw new IllegalArgumentException("Unknown account type ${credentials.accountType}")
-    }
-    provider
+    OpenstackIdentityProvider identityProvider = new OpenstackIdentityV3Provider(credentials)
```

Comment:
That's the version we are supporting, as it ships with liberty and is the current version. In the future we may upgrade to v4, if it comes out. The name just makes it explicit as to which version we are using.